query/src/main/java/org/infinispan/query/indexmanager/LuceneWorkTransformationVisitor.java

```diff
@@ -33,33 +34,38 @@ private LuceneWorkTransformationVisitor() {
    }
 
    @Override
-   public LuceneWorkIdTransformer getDelegate(AddLuceneWork addLuceneWork) {
-      return addDuplicator;
+   public LuceneWork visitAddWork(AddLuceneWork addLuceneWork, KeyTransformationHandler keyTransformationHandler) {
+      return addDuplicator.cloneOverridingIdString( addLuceneWork, keyTransformationHandler );
    }
 
    @Override
-   public LuceneWorkIdTransformer getDelegate(DeleteLuceneWork deleteLuceneWork) {
-      return deleteDuplicator;
+   public LuceneWork visitDeleteWork(DeleteLuceneWork deleteLuceneWork, KeyTransformationHandler keyTransformationHandler) {
+      return deleteDuplicator.cloneOverridingIdString( deleteLuceneWork, keyTransformationHandler );
    }
 
    @Override
-   public LuceneWorkIdTransformer getDelegate(UpdateLuceneWork updateLuceneWork) {
-      return updateDuplicator;
+   public LuceneWork visitUpdateWork(UpdateLuceneWork updateLuceneWork, KeyTransformationHandler keyTransformationHandler) {
+      return updateDuplicator.cloneOverridingIdString( updateLuceneWork, keyTransformationHandler );
    }
 
    @Override
-   public LuceneWorkIdTransformer getDelegate(OptimizeLuceneWork optimizeLuceneWork) {
-      return returnSameDuplicator;
+   public LuceneWork visitOptimizeWork(OptimizeLuceneWork optimizeLuceneWork, KeyTransformationHandler keyTransformationHandler) {
+      return returnSameDuplicator.cloneOverridingIdString( optimizeLuceneWork, keyTransformationHandler );
    }
 
    @Override
-   public LuceneWorkIdTransformer getDelegate(FlushLuceneWork flushLuceneWork) {
-      return returnSameDuplicator;
+   public LuceneWork visitFlushWork(FlushLuceneWork flushLuceneWork, KeyTransformationHandler keyTransformationHandler) {
+      return returnSameDuplicator.cloneOverridingIdString( flushLuceneWork, keyTransformationHandler );
    }
 
    @Override
-   public LuceneWorkIdTransformer getDelegate(PurgeAllLuceneWork purgeAllLuceneWork) {
-      return returnSameDuplicator;
+   public LuceneWork visitPurgeAllWork(PurgeAllLuceneWork purgeAllLuceneWork, KeyTransformationHandler keyTransformationHandler) {
+      return returnSameDuplicator.cloneOverridingIdString( purgeAllLuceneWork, keyTransformationHandler );
+   }
+
+   @Override
+   public LuceneWork visitDeleteByQueryWork(DeleteByQueryLuceneWork work, KeyTransformationHandler p) {
+	   throw new UnsupportedOperationException( "delete-by-query is not supported" );
```

Comment:
As today there's no use for it; we either delete by id or Class, this might come handy for the cases when someone is using a custom ```SearchWorkCreator``` that generates many extra documents in the index during insertions to be able to efficiently delete them afterwards.